store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/ApplyViewToElementsFunction.java

```diff
@@ -77,13 +84,31 @@ public ApplyViewToElementsFunction(final Map<String, Object> context) throws Gaf
             }
             // Validate the supplied context before using
             validate(context);
+
+            updateViewWithValidationFromSchema(context);
+
             this.context = Collections.unmodifiableMap(context);
         } catch (final Exception e) {
             throw new GafferCheckedException("Unable to create TemporaryResultsGraph", e);
         }
 
     }
 
+    private static void updateViewWithValidationFromSchema(final Map<String, Object> context) {
```

Comment:
Should the function `ApplyViewToElementsFunction` be renamed?

Reply:
To what?

It still applies view to elements. This additional new Validation suff is because of MapStore failings, this would be done for free otherwise. It should get ripped out when Mapstore is fixed.

Reply:
It doesn't just apply the view anymore, it also reapplies the schema now. On this topic, does your fix apply schema aggregation too? Or just schema validation?

Reply:
"does your fix apply schema aggregation too?" - I think this is not necessary as the MapStore does this.

Reply:
yep correct, mapstore does it. This is all due to Mapstore having issues that need completing.

Reply:
Perhaps this could actually be a different function called `ApplyViewAndSchemaToElementsFunction` and be an alternative?

Reply:
I think we landed on "MergeElementFunction"